src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userByEmail = userService.findByEmail(email);
+        if (userByEmail.isEmpty() || !HashUtil.getHash(password, userByEmail.get().getSalt())
+                .equals(userByEmail.get().getPassword())) {
+            throw new AuthenticationException("Can't authenticate User");
```

Comment:
```suggestion
            throw new AuthenticationException("Email or password is incorrect");
```
this is the message user going to see on ui, so let's be more informative